daffodil-io/src/main/scala/org/apache/daffodil/io/DirectOrBufferedDataOutputStream.scala

```diff
@@ -278,42 +296,23 @@ class DirectOrBufferedDataOutputStream private[io] (
    * be completely configured (byteOrder, encoding, bitOrder, etc.)
    */
   def addBuffered(): DirectOrBufferedDataOutputStream = {
-    val buffered = new DirectOrBufferedDataOutputStream(this, isLayer)
+    val buffered = new DirectOrBufferedDataOutputStream(
+      this,
+      isLayer,
+      chunkSizeInBytes,
+      maxBufferSizeInBytes,
+      tempFilePath)
     addBufferedDOS(buffered)
     buffered
   }
 
-  def addBufferedBlob(
+  def addBlobFile(
     path: Path,
     lengthInBits: Long,
     blobChunkSizeInBytes: Int,
-    finfo: FormatInfo): DirectOrBufferedDataOutputStream = {
+    finfo: FormatInfo): Unit = {
 
-    // create a special buffered blob data outputstream and split the current
-    // DOS to it. When this normal DOS is finished, the blob DOS will handle
-    // delivering the blob data to it without loading the whole blob into
-    // memory all at once.
-    val bufferedBlob = new BufferedBlobDataOutputStream(
-      this,
-      path,
-      lengthInBits,
-      blobChunkSizeInBytes,
-      isLayer)
-    addBufferedDOS(bufferedBlob)
-
-    // we know the length of the blob as passed in, so adjust the bit position
-    // and mark it as finished.
-    bufferedBlob.setRelBitPos0b(bufferedBlob.relBitPos0b + ULong(lengthInBits))
-    if (lengthInBits > 0)
-      bufferedBlob.setNonZeroLength()
-    bufferedBlob.setFinished(finfo)
-
-    // now split the blob DOS to a normal buffered DOS, we'll return this and
-    // expect the caller to update the UState accordingly so that all future
-    // unparsed data is written to this DOS. This DOS will eventually be
-    // delivered to the blob DOS once it becomes direct
-    val buffered = bufferedBlob.addBuffered()
-    buffered
+    this.putFile(path, lengthInBits, blobChunkSizeInBytes, finfo)
```

Comment:
I'm not sure this is correct. This will cause Daffodil to write the entire contents of the blob to the current data output stream. If the current DOS buffered this will write the full contents to that buffer. Eventually that buffer will fill up and switch to file based and write the entire content to a file, and then we'll finish writing to the blob file to that other file.

So this could be very inefficient if the DOS is buffered.

Instead, I still think we want similar logic as existed before where we split to two DOS, but instead of creating a special BufferedBlobDataOutputStream we want to create a DirectOrBufferedDataOutputstream where the ByteArrayOrFIleOutputStream never uses a ByteArrayOutputStream and goes straight to the FileOutputStream. And the FileOutputStream should be the backed by the blob file instead of a temporary file.